flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-hudi/src/main/java/org/apache/flink/cdc/connectors/hudi/sink/util/RowDataUtils.java

```diff
@@ -0,0 +1,604 @@
+
+        RecordData recordData;
+        RowKind kind;
+        switch (dataChangeEvent.op()) {
+            case INSERT:
+            case UPDATE:
+            case REPLACE:
+                {
+                    recordData = dataChangeEvent.after();
+                    kind = RowKind.INSERT;
+                    break;
+                }
+            case DELETE:
+                {
+                    recordData = dataChangeEvent.before();
+                    kind = RowKind.DELETE;
+                    break;
+                }
+            default:
+                throw new IllegalArgumentException("don't support type of " + dataChangeEvent.op());
+        }
+        GenericRowData genericRowData = new GenericRowData(recordData.getArity());
+        genericRowData.setRowKind(kind);
+        for (int i = 0; i < recordData.getArity(); i++) {
+            genericRowData.setField(i, fieldGetters.get(i).getFieldOrNull(recordData));
+        }
+        return genericRowData;
+    }
+
+    public static List<FieldGetter> createFieldGetters(Schema schema, ZoneId zoneId) {
+        List<Column> columns = schema.getColumns();
+        List<FieldGetter> fieldGetters = new ArrayList<>(columns.size());
+        for (int i = 0; i < columns.size(); i++) {
+            fieldGetters.add(createFieldGetter(columns.get(i).getType(), i, zoneId));
+        }
+        return fieldGetters;
+    }
+
+    /** Create a {@link FieldGetter} for the given {@link DataType}. */
+    public static FieldGetter createFieldGetter(DataType fieldType, int fieldPos, ZoneId zoneId) {
+        final FieldGetter fieldGetter;
+        // ordered by type root definition
+        switch (fieldType.getTypeRoot()) {
+            case CHAR:
+            case VARCHAR:
+                fieldGetter =
+                        row ->
+                                org.apache.flink.table.data.StringData.fromString(
+                                        row.getString(fieldPos).toString());
+                break;
+            case BOOLEAN:
+                fieldGetter = row -> row.getBoolean(fieldPos);
+                break;
+            case BINARY:
+            case VARBINARY:
+                fieldGetter = row -> row.getBinary(fieldPos);
+                break;
+            case DECIMAL:
+                final int decimalScale = DataTypeChecks.getScale(fieldType);
+                int precision = getPrecision(fieldType);
+                fieldGetter =
+                        row -> {
+                            DecimalData decimalData =
+                                    row.getDecimal(fieldPos, precision, decimalScale);
+                            return org.apache.flink.table.data.DecimalData.fromBigDecimal(
+                                    decimalData.toBigDecimal(), precision, decimalScale);
+                        };
+                break;
+            case TINYINT:
+                fieldGetter = row -> row.getBoolean(fieldPos);
```

Comment:
`getByte(fieldPos)`?

Reply:
Same underlying implementation with more validation, let's keep it.